src/diffusers/models/embeddings.py

```diff
@@ -78,6 +78,53 @@ def get_timestep_embedding(
     return emb
 
 
+def aryan_get_3d_sincos_pos_embed(
+    embed_dim: int,
+    spatial_size: Union[int, Tuple[int, int]],
+    temporal_size: int,
+    spatial_interpolation_scale: float = 1.0,
+    temporal_interpolation_scale: float = 1.0,
+    device: Optional[torch.device] = None,
```

Comment:
From the code, where we're calling these functions, I think it would be useful to always pass the right device. That way, we won't have to incur any device placement costs.

Reply:
Yes ofcourse, sounds good!